sui/src/microbench.rs

```diff
@@ -292,6 +297,59 @@ impl ClientServerBenchmark {
             Duration::from_micros(self.send_timeout_us),
             Duration::from_micros(self.recv_timeout_us),
         );
+
+        // We spawn a second client that listens to the batch interface
+        let client_batch = NetworkClient::new(
+            self.host.clone(),
+            self.port,
+            self.buffer_size,
+            Duration::from_micros(self.send_timeout_us),
+            Duration::from_micros(self.recv_timeout_us),
+        );
+
+        let _batch_client_handle = tokio::task::spawn(async move {
+            let authority_client = AuthorityClient::new(client_batch);
+
+            let mut start = 0;
+
+            loop {
+                let receiver = authority_client
+                    .handle_batch_streaming_as_stream(BatchInfoRequest {
+                        start,
+                        end: start + 10_000,
+                    })
+                    .await;
+
+                if let Err(e) = &receiver {
+                    error!("Listener error: {:?}", e);
+                    break;
+                }
+                let mut receiver = receiver.unwrap();
+
+                info!("Start batch listener at sequence: {}.", start);
+                while let Some(item) = receiver.next().await {
+                    match item {
+                        Ok(BatchInfoResponseItem(UpdateItem::Transaction((
+                            _tx_seq,
+                            _tx_digest,
+                        )))) => {
+                            start = _tx_seq + 1;
+                        }
```

Comment:
I have run this locally and confirmed that it is functioning as a stream now. 👍 

We are only generating transaction load and not batch load so far, is that right or are we making batches somewhere, as alluded to in the title?